app/src/main/java/eu/kanade/tachiyomi/data/library/LibraryUpdateJob.kt

```diff
@@ -320,6 +316,36 @@ class LibraryUpdateJob(private val context: Context, workerParams: WorkerParamet
         }
     }
 
+    /**
+     * Processes new chapters for a given manga and handles related updates, such as downloading
+     * new chapters and updating the notification states.
+     *
+     * @param manga The manga for which new chapters are being handled.
+     * @param newChapters A list of new chapters for the manga.
+     * @param chaptersToDownload A list of chapters to be downloaded based on user preferences.
+     * @param newUpdates A thread-safe list that stores pairs of manga and their respective new chapters.
+     * @param hasDownloads A flag indicating whether there are any chapters to download.
+     */
+    private suspend fun handleNewChapters(
+        manga: Manga,
+        newChapters: List<Chapter>,
+        chaptersToDownload: List<Chapter>,
+        newUpdates: CopyOnWriteArrayList<Pair<Manga, Array<Chapter>>>,
+        hasDownloads: AtomicBoolean
+    ) {
+        val categoryIds = getCategories.await(manga.id).map { it.id }
+        if (manga.shouldDownloadNewChapters(categoryIds, downloadPreferences) && chaptersToDownload.isNotEmpty()) {
```

Comment:
Should I also call `manga.shouldDownloadNewChapters(categoryIds, downloadPreferences)` or move the logic of this extension in this function to `SyncChaptersWithSource`?

It's only used in this Class and MangaScreenModel.kt

Reply:
Make a new interactor to determine what chapters to download given the manga and new chapters and move out the logic from `SyncChaptersWithSource`

Reply:
Do you want `SyncChaptersWithSource` to return a `Pair` of new chapters and chapters to download by calling the new interactor? Or would you prefer that SyncChaptersWithSource only returns the new chapters, leaving the determination of chapters to download to be handled separately?

Reply:
Nah revert the changes in `SyncChaptersWithSource` but use the new logic inside it in the new interactor